### ISSUE TITLE
[istio-proxy] envoyMetricsService should work with auto mTLS

### ISSUE THREAD

**Issue by PatrickDuncan** [istio/istio#26968]
**Describe the feature request**

https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#ProxyConfig
The `defaultConfig.envoyMetricsService` DNS resolves the address to an IP. It then uses that IP to make outbound requests. To use TLS, it has `tlsSettings` which is a DestinationRule under the covers.

This is not compatible with mTLS between pods in a mesh. The envoy doesn't know that the IP is a part of the mesh (it doesn't use the outbound rule). The requests return a termination error:
```yaml
spec:
  meshConfig:
    defaultConfig:
      envoyMetricsService:
        address: my-metrics-service:9818
```

```
':method', 'POST'
':path', '/envoy.service.metrics.v2.MetricsService/StreamMetrics'
':authority', 'envoy_metrics_service'
':scheme', 'http'
'te', 'trailers'
'content-type', 'application/grpc'
'x-envoy-internal', 'true'
'x-forwarded-for', '172.30.112.125'
```

```
'grpc-status', '14'
'grpc-message', 'upstream connect error or disconnect/reset before headers. reset reason: connection termination'
```

---

[grpc curling](https://github.com/fullstorydev/grpcurl) a metrics request from a go pod to my metricsService succeeds and it is auto converted to mTLS. This works because I'm using the hostname directly, not the IP of the metricsService k8s service.

```shell
grpcurl -d '{"identifier": {"node": {}}, "envoy_metrics": []}' -plaintext my-metrics-service:9818 envoy.service.metrics.v2.MetricsService/StreamMetrics
```

```
'grpc-status', '0'
'grpc-message', ''
```

```
':method', 'POST'
':scheme', 'http'
':path', '/envoy.service.metrics.v2.MetricsService/StreamMetrics'
':authority', 'my-metrics-service:9818'
'content-type', 'application/grpc'
'user-agent', 'grpc-go/1.33.0-dev'
'te', 'trailers'
```

```
':method', 'POST'
':scheme', 'https'
':path', '/envoy.service.metrics.v2.MetricsService/StreamMetrics'
':authority', 'my-metrics-service:9818'
'content-type', 'application/grpc'
'user-agent', 'grpc-go/1.33.0-dev'
'te', 'trailers'
'x-forwarded-proto', 'http'
'x-request-id', 'b3bd7f8f-fa39-426b-ba11-e94cc3e95a06'
'x-envoy-decorator-operation', 'my-metrics-service.default.svc.cluster.local:9818/*'
'x-envoy-peer-metadata', '<REDACTED>'
'x-envoy-peer-metadata-id', 'sidecar~172.30.112.125~sleep-56dd55d645-pgkjj.default~default.svc.cluster.local'
'x-envoy-attempt-count', '1'
'x-b3-traceid', '48c8d97bc52576c71b74be441cf9cc6f'
'x-b3-spanid', '1b74be441cf9cc6f'
'x-b3-sampled', '0'
```

---

My actual feature request is to make envoyMetricsService requests use the hostname provided in the address, instead of resolving it and using the IP.

This should also be a problem with tracing and `envoyAccessLogService`

**Describe alternatives you've considered**

I have a special use case that needs this kind of implicit mTLS

**Affected product area (please put an X in all that apply)**

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by bianpengyuan** [comment]
I think you'll need to set ClientTLSSettings to make it use mTLS?

**Comment by bianpengyuan** [comment]
https://github.com/istio/api/blob/c78f387f78a22a3c4a1c49679dc44cded454530d/mesh/v1alpha1/proxy.proto#L459

**Comment by PatrickDuncan** [comment]
@bianpengyuan Hey, ya that does work normally. In my case I'm using SNI passthrough to send metrics from the sidecar to my metrics service in a different cluster. I've added the metrics service to the mesh by adding a sidecar (`istioctl proxy-status` shows that it's in sync).

I did try using ClientTLSSettings with SNI passthrough, but I run into a `connection failure`. If the envoyMetricsService resolves the hostname to an IP, then the SNI value would be stripped away. Different problem, same root cause I think.
